src/main/java/com/inventoryservice/controller/CategoryController.java

```diff
@@ -0,0 +1,187 @@
+    public ResponseEntity<?> updateCategory(
+            @PathVariable int categoryId,
+            @RequestBody Category category
+    ) throws CategoryNotFoundException {
+        Category categories = categoryService.findCategoryById(categoryId);
+        if (categories == null) {
+            throw new CategoryNotFoundException("Category not found");
+        }
+        if (!categories.isArchived()) {
+            Category updatedCategory = categoryService.updateCategory(categoryId, category);
+            return ResponseEntity.ok(updatedCategory);
+        }
+        return new ResponseEntity<>("Category is archived ", HttpStatus.BAD_REQUEST);
+    }
+
+    @PostMapping("/{categoryId}/subCategory")
+    @PermitInventoryManager
+    public ResponseEntity<List<?>> addSubCategories(
+            @PathVariable("categoryId") int categoryId,
+            @RequestBody List<SubCategoryRequest> subCategoryRequests) {
+        Category category = categoryService.findCategoryById(categoryId);
+        List<SubCategory> subCategories = subCategoryRequests.stream()
+                .map(subCategoryRequest -> {
+                    SubCategory subCategory = SubCategoryRequest.toEntity(subCategoryRequest);
+                    subCategory.setCategory(category);
+                    return subCategory;
+                })
+                .collect(Collectors.toList());
+        List<SubCategory> savedSubCategories = subcategoryService.saveSubcategories(subCategories);
+        return ResponseEntity.status(HttpStatus.CREATED).body(savedSubCategories);
+    }
+
+    /*
+     * To get list of all SubCategory
+     * @return      list of all SubCategory .
+     */
+    @GetMapping(("/subCategory"))
+    public ResponseEntity<List<SubCategory>> getAllSubCategory() {
+        List<SubCategory> subCategories = subcategoryService.getAllSubCategory();
+        return new ResponseEntity<>(subCategories, HttpStatus.OK);
+    }
+
+    /*  To Delete subCategory
+     * @param        subCategoryId an Id giving the subCategoryId of specific Id
+     * @return      it deleted subCategoryId of specific Id.
+     * @throws SubCategoryNotFoundException  If an subCategory of specific Id is not found
+     */
+    @DeleteMapping("/{categoryId}/subCategory/{subCategoryId}")
+    @PermitInventoryManager
+    public ResponseEntity<?> deleteSubCategory(@PathVariable int subCategoryId) throws SubCategoryNotFoundException {
+        SubCategory subCategory = subcategoryService.findSubCategoryById(subCategoryId);
+        if (subCategory == null) {
+            throw new SubCategoryNotFoundException("subCategory not found");
+        }
+        if (!subCategory.isArchived()) {
+            subcategoryService.deleteSubCategory(subCategoryId);
+            return new ResponseEntity<String>("subCategory Deleted successfully", HttpStatus.FOUND);
+        }
+        return new ResponseEntity<String>("subCategory is archived and cannot be deleted", HttpStatus.BAD_REQUEST);
+    }
```

Comment:
Wrong implementation.         if (subCategory.isArchived()) {
 throw Exception here (send below response in exception)
     new ResponseEntity<String>("subCategory is archived and cannot be deleted", HttpStatus.BAD_REQUEST);
           
        }
 subcategoryService.deleteSubCategory(subCategoryId);
            return new ResponseEntity<String>("subCategory Deleted successfully", HttpStatus.FOUND);

Reply:
Completed